core/src/main/java/hudson/model/Run.java

```diff
@@ -465,6 +465,17 @@ public void setResult(@Nonnull Result r) {
             LOGGER.log(FINE, this + " in " + getRootDir() + ": result is set to " + r, LOGGER.isLoggable(Level.FINER) ? new Exception() : null);
         }
     }
+    
+    /**
+     * Forced to set the {@link #getResult} of this build even when result is already set.
+     * May only be called after the build has started.
+     * @param r the proposed new result
+     */
```

Comment:
Needs <code>@since TODO</code>. Probably it makes sense to mark the method as deprecated to highlight that it is not recommended (excepting particular cases)